requirements/base.txt

```diff
@@ -1,17 +1,13 @@
-os-client-config==1.28.0
-osc-lib==1.2.0
-cliff==2.2.0
-oslo.utils==3.37.1
-oslo.serialization==2.28.1
-keystoneauth1==3.13.0
-python-keystoneclient==3.17.0
-python-heatclient==1.6.1
-python-novaclient==7.1.2
-google-api-python-client==1.7.7
-paramiko==2.1.6
-celery==3.1.18
-apscheduler==3.5.1
-jsonfield==2.0.2
-six==1.11.0
--e git+https://github.com/edx/xblock-utils.git@v1.0.5#egg=xblock-utils==v1.0.5
+apscheduler>=3.5.1,<3.6
```

Comment:
No real objection here, just curious why all of these do specify maximum versions.

Reply:
The idea is to have a relaxed set of requirements in setup.py for maximum runtime flexibility (an attempt at reducing last-minute fixes when some pip dependency gets upgraded), but a more reproducible environment in requirements.txt - who knows if a future Heat version will be incompatible with the current API.  A point release freeze would work, but then we wouldn't get the odd security/bug fix without manual work.  Hence, a middle ground with maximum versions.